source/manual/conventions-in-rails-applications.html.md

```diff
@@ -0,0 +1,301 @@
+to be the prevalent convention where:
+
+- `/app` is used for:
+  - the default Rails components (such as controllers, models, views, helpers, jobs)
+  - commonly used classes in the application that have a common purpose and
+    interface (such as services or policies)
+- `/lib` is used for:
+  - classes that don't logically fit into an app directory
+  - complex business logic for the application
+  - Rake tasks
+
+GOV.UK Rails applications should avoid creating an `/app/lib` directory, as this
+causes confusion with the prevalent `/lib` directory.
+
+Configuration for applications, typically [YAML][] files, should live in the
+`/config` directory.
+
+[app-lib]: https://medium.com/extreme-programming/what-goes-in-rails-lib-92c74dfd955e
+[YAML]: https://yaml.org/
+
+## Documenting your decisions
+
+It's conventional to create a `/docs` directory in GOV.UK Rails applications to
+store application documentation (which is typically linked to from the readme).
+
+For recording the reasoning behind architectural decisions it's a conventional
+to use [architectural decision records][adr] in the `/docs/adr` directory
+([example][adr-example]).
+
+[adr]: http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions
+[adr-example]: https://github.com/alphagov/content-publisher/blob/master/docs/adr/0002-use-local-datastore.md
+
+## Editing of microcopy by non-developers
+
+When it is intended for non-developers to edit microcopy in an application it is
+conventional to use [Rails Internationalization (I18n)][i18n] to abstract
+the copy outside of the application code. We use this even when an application
+is only in English. These files are stored in `/config/locales` as per Rails
+convention. You should be cautious to only these for accessing copy and not
+for any application configuration.
+
+[i18n]: https://guides.rubyonrails.org/i18n.html
+
+## Error handling
+
+GOV.UK Rails applications should anticipate failure and do so gracefully. This
+[happens automatically, to a degree, on GOV.UK frontend
+applications][frontend-errors] but not on admin applications where there is more
+granular control over error responses. GOV.UK Rails applications should be
+configured to provide GOV.UK branded error responses.
+[Content Publisher][cp-errors]  provides an example of implementing this with
+the [GOV.UK admin layout][].
+
+[frontend-errors]: /manual/errors.html
+[cp-errors]: https://github.com/alphagov/content-publisher/pull/1252
+[GOV.UK admin layout]: https://components.publishing.service.gov.uk/component-guide/layout_for_admin
+
+## Testing strategies
+
+We aim for well tested Rails applications that provide a good degree of coverage
+and provide assurances that an application is operating correctly. We value
+test suites that are fast to run, easy to read and [test things consistently at
+different abstractions (unit, integration and functional)][cp-testing-strategy].
+
+### Common tooling
```

Comment:
```suggestion
### Common tooling
```

Reply:
Ooh good catch - where do they come from?

Reply:
It happens to me a lot too! It's the result of `ALT + SPACE`, which is common because the `ALT` key is often still being held down after pressing the 3 key to make a `#` for the heading.

Reply:
Oh thank you! TIL

Reply:
I've got this in my `.vimrc` to auto-replace them: https://github.com/issyl0/dotfiles/blob/master/.vimrc#L42-L45

Reply:
Ooh pro tip 👍